app/models/sales_log.rb

```diff
@@ -171,6 +172,10 @@ def mortgage_not_used?
     mortgageused == 2
   end
 
+  def set_mortgage_value_zero
+    self.mortgage = 0
+  end
```

Comment:
I was looking at this for a while with Nat this morning. Perhaps you know something that we don't that will let us do this.
We first wanted to put it in derived variables - but if mortgageused == 2 (no), then mortgage value question is not routed, therefore on the line after set_derived_fields! we hit reset_invalidated_dependent_fields! inside of which as far as I can see mortgage is then reset to nil as it is no longer routed to.
We tried changing the order of those two methods because on a vague feeling level it seems like they shouldn't mind that, but according to a handful of tests they do mind that (we didn't investigate those tests and why they are failing at this point but could revisit that)

Reply:
You could try changing `selected_answer_option_is_derived?` question method specifically for that question. I think we do it for some other questions already

Reply:
Could you not set it to zero anyway, even if it's previously made nil?